clang/test/AST/HLSL/ByteAddressBuffer-AST.hlsl

```diff
@@ -0,0 +1,41 @@
+// RUN: %clang_cc1 -triple dxil-pc-shadermodel6.0-library -x hlsl -ast-dump -DEMPTY %s | FileCheck -check-prefix=EMPTY %s
+// RUN: %clang_cc1 -triple dxil-pc-shadermodel6.0-library -x hlsl -ast-dump %s | FileCheck %s
+
```

Comment:
FYI, we are planning to refactor the resource buffer AST tests, see here: https://github.com/llvm/llvm-project/pull/113643#discussion_r1833370629

Maybe there could be a single AST test with multiple run lines that would tests all of the raw buffer combinations using defines and different check-prefixes. Do you want to update this test to be ready for adding the next raw buffer variants or wait until the next buffer class is added?

Reply:
I refactor the test a little to make it more generic. However, I feel more comfortable making this change once we have more buffer classes, this way I will have more code to create patterns of.

Reply:
I think it would be better if you commit the test in the original form and update it to cover more buffer types once you have more code the see the patterns. I have some ideas how to do it, let's chat later.